src/components/profile-page/access-tokens/access-token-creation-form.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import type { ChangeEvent, FormEvent } from 'react'
+import React, { Fragment, useCallback, useEffect, useMemo, useState } from 'react'
+import { Button, Form } from 'react-bootstrap'
+import { cypressId } from '../../../utils/cypress-attribute'
+import { Trans, useTranslation } from 'react-i18next'
+import { AccessTokenCreatedModal } from './access-token-created-modal'
+import { DateTime } from 'luxon'
+import type { AccessTokenWithSecret } from '../../../api/tokens/types'
+import { postNewAccessToken } from '../../../api/tokens'
+import { showErrorNotification } from '../../../redux/ui-notifications/methods'
+
+/**
+ * Form for creating a new access token.
+ */
+export const AccessTokenCreationForm: React.FC = () => {
+  const { t } = useTranslation()
+  const [newTokenLabel, setNewTokenLabel] = useState('')
+  const [newTokenExpiry, setNewTokenExpiry] = useState('')
+  const [newTokenWithSecret, setNewTokenWithSecret] = useState<AccessTokenWithSecret>()
+  const [showCreatedModal, setShowCreatedModal] = useState(false)
+
+  const onHideCratedModal = useCallback(() => {
+    setShowCreatedModal(false)
+  }, [])
+
+  const onChangeTokenLabel = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setNewTokenLabel(event.target.value)
+  }, [])
+
+  const onChangeTokenExpiry = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setNewTokenExpiry(event.target.value)
+  }, [])
+
+  const onCreateToken = useCallback(
+    (event: FormEvent<HTMLFormElement>) => {
+      event.preventDefault()
+      postNewAccessToken(newTokenLabel, newTokenExpiry)
+        .then((tokenWithSecret) => {
+          setNewTokenWithSecret(tokenWithSecret)
+          setShowCreatedModal(true)
+        })
+        .catch(showErrorNotification('profile.accessTokens.creationFailed'))
+    },
+    [newTokenLabel, newTokenExpiry]
+  )
+
+  const newTokenCreatable = useMemo(() => {
+    return newTokenLabel.trim() !== ''
+  }, [newTokenLabel])
+
+  const expiryDates = useMemo(() => {
+    const today = DateTime.now()
+    return {
+      min: today.toISODate(),
+      max: today
+        .plus({
+          year: 2
+        })
+        .toISODate(),
+      default: today
+        .plus({
+          year: 1
+        })
+        .toISODate()
+    }
+  }, [])
+
+  useEffect(() => {
+    setNewTokenExpiry(expiryDates.default)
```

Comment:
Why aren't you using the default function of the useState that you're manipulating here?